activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1108,6 +1108,17 @@ def arel_columns(columns)
     end
 
     def reverse_sql_order(order_query)
+      if offset_index != 0
+        ActiveSupport::Deprecation.warn(<<-WARNING.squish)
+            Finding a last element by loading the relation when SQL ORDER
```

Comment:
We are adding the deprecation warning only when calling `order` with `offset`, so should we update the deprecation message to only talk about `Finding last element on a relation with offset applied is deprecated` ?